dist/rpm/singularity.spec.in

```diff
@@ -45,130 +45,65 @@ Requires: squashfs
 Requires: squashfs-tools
 %endif
 
-Requires: %{name}-runtime = %{version}-%{release}
+
+Requires: %{name}-unpriv = %{version}-%{release}
+Provides: %{name}-runtime
+Obsoletes: %{name}-runtime
 
 %description
 Singularity provides functionality to make portable
 containers that can be used across host environments.
 
-%package devel
-Summary: Development libraries for Singularity
-Group: System Environment/Development
-
-%description devel
-Development files for Singularity
-
-%package runtime
-Summary: Support for running Singularity containers
+%package unpriv
```

Comment:
In our distro released SPEC file, this was going to be a single RPM package. A very simple, basic, and _example_ SPEC file. 

These are decisions to be made downstream by the package maintainers, not by us. They should be in control of the SPEC file they use and not using ours.

Are you willing to change it so this generates a _single_ RPM package?

Reply:
I am the primary downstream package maintainer, for EPEL.  My preference is to make it be as close to the upstream as possible, so that people who compile from upstream source will get as similar an experience as possible.  So I would change it to a single package if I must, but I would rather not, because EPEL will have the dual package.

Reply:
The reason for introducing the dual package is now that RHEL 7.6 is going to have full support for unprivileged mount namespaces, so having a separate unprivileged package makes sense.

Reply:
> My preference is to make it be as close to the upstream as possible

That's fine. But there are things that you _want_ that don't fit in with what we want upstream. We don't _want_ to be in packaging. We want just a simple example that is usable, if wanted, but doesn't try to be the end all position on how it should be. It's an _example_ that can be used to flesh out a more site specific version if wanted by that admin.

> so having a separate unprivileged package makes sense

If you're doing that, I think you should have reversed it. So instead you have a `package priv` (i.e. singularity-priv) and that includes only the `starter-suid` binary. Everything else is in the main singularity package. How you have it, when you do:

As it is in the PR, if you do `yum install singularity` ... you're installing the *SUID* component. You'd need `yum install singularity-unpriv` for the main Singularity binary. Not what you would expect to happen.

And again, from our point of view. We want out of the packaging market. We want a bare bones and simple example file. It builds an RPM. Nothing fancy, but is a base upon which others can expand on if they wish.

Reply:
Could the singularity core team look at it this way: let @drdaved maintain the example spec file in the distribution?  You don't have to worry about it; let any questions & reviews go to me.  EPEL is by far the most widely used distribution of rpm so I think it makes sense to make it be the base-level example.  Site admins can modify that as they like then.  My concern with not doing that and including it in travis builds is that packaging frequently gets broken during the development process.  I experienced that many times with 2.x before singularity travis started building rpms.

With the PR as it is, if someone does `yum install singularity` they automatically get both packages, because the singularity package requires singularity-unpriv of the same version number.  I don't think it's a good idea to put the suid binary only in a singularity-priv package because then some types of basic functionality will not work with just the singularity package.  I think the name "singularity" should contain the full suite, or there will be lots of support questions from people who are just starting out.  Let the people who know that they can live without the suid binary go through the extra effort to learn to use a different package name.

Reply:
>Could the singularity core team look at it this way: let @DrDaveD maintain the example spec file in the distribution? You don't have to worry about it; let any questions & reviews go to me.

I think that is a great idea, but I think the spec file(s) belong(s) in another repo where @DrDaveD has the ability to merge.  

Only Sylabs-ers have merge permissions in this repo and that is not going to change anytime soon.  So that means Sylabs-ers must review anything that goes into this repo.  We have a policy of 2 reviews per merge.  That is why we don't want complicated set spec files living here and frequent packaging PRs coming in.  It's just too big an investment of time.  

@DrDaveD why not create another repo where these and other spec files could live?  And Singularity docs could point folks there?

Reply:
@DrDaveD ... if you're willing to remove the `package unpriv` here in this PR, I'm willing to approve it (and wait for @yhcote to speak up on the makeit parts).

From my point of view, overall it *is* a good PR. It's just that splitting of the RPM package that we're really not wanting to have in the upstream version that's causing the biggest push back from me.

Reply:
Ok I removed the unpriv packaging.

Reply:
Thank you very much.